src/main/resources/db/changelog/changes/00000000000000_initial_schema.xml

```diff
@@ -11,10 +11,29 @@
     <property name="autoIncrement" value="true"/>
 
     <changeSet id="00000000000000" author="nivethika@thehyve.nl" dbms="postgresql,oracle,h2">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <sequenceExists sequenceName="hibernate_sequence"/>
```

Comment:
Did  you try this with connection to database which had older version of schema.
Ideally, you should add new changelog, instead of changing existing change-set. This will mofify the existing offset created for offset.

Reply:
Hi yes i have tested this with the old database and works fine.
I changed the existing change-set because in any case liquibase picks it up as a new change set (because of change in checksum logic) so I guess it should not have any adverse effect. Plus i am not sure how to use preconditions on an existing change-set using a new change-set (another option is validating the old change-set checksum which i am not sure is ideal).

Reply:
@nivemaham do you have any suggestions ?

Reply:
If it works fine with me. What dose "MARK_RAN" mean?

Reply:
https://www.liquibase.org/documentation/existing_project.html according to this, preconditions impact negatively on performance.

Reply:
`MARK_RAN` will mark the changeset as ran on the failure of the pre-condition (so in this case if a table/index already exists) so Liquibase doesn't try to run it every time.

> https://www.liquibase.org/documentation/existing_project.html according to this, preconditions impact negatively on performance.

Yes, there is a penalty but it's only on the first run. As after that liquibase knows that it has run (hence the `MARK_RAN`).

> this would avoid creating the hibernate_sequence on a fresh installation if i understand it correctly

No, it will only fail if the sequence already exists. If the sequence does not exist, the pre-condition will be successful and the change-set will be executed.

Reply:
OK. Thanks for the info. Still something is wrong. I do not understand how liquibase fails only on createSequence adn createIndex, but not createTable.

Reply:
It failed on createTable too for me.

Reply:
in addition, I propose we migrate to the use of  
```
  - include: file name
```
instead of includeAll.

Reply:
> It failed on createTable too for me.

My bad. I didn't check the changelog well. if it fails for everything, then I think how we define the change-log could be a problem. I will have a look

Reply:
I think the expected behavior should be that the liquibase should run this changeset only once. if the changeset is modified, the check-sum changes, thus it should fail. This can be used as a workaround. but when we add new entries to the change-set we will encounter this issue again